front-end/src/app/login/guards/login-page.guard.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router, UrlTree } from '@angular/router';
+import { ApiService } from 'app/shared/services/api.service';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LoginGuard implements CanActivate {
+  constructor(private apiService: ApiService, private router: Router) {}
+  canActivate(): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    if (this.apiService.isAuthenticated()) {
+      this.router.navigate(['dashboard']);
```

Comment:
Navigate is [asynchronous](https://github.com/angular/angular/blob/88e16553e88356339db4fbc6aa1b78a9dfdfc8d2/packages/router/src/router.ts#L1215), could you add a return here so that in the case where navigate errors, the user isn't passed through the guard